src/FakeItEasy.Extensions.ValueTask/FakeItEasy.Extensions.ValueTask.csproj

```diff
@@ -0,0 +1,62 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net45;netstandard1.6;netstandard2.0;netstandard2.1</TargetFrameworks>
+    <SignAssembly>true</SignAssembly>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)' == 'Debug'">
+    <DebugType>full</DebugType>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)' == 'Release'">
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <DebugType>pdbonly</DebugType>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'netstandard2.1'">
+    <Compile Include="../FakeItEasy/ValueTaskReturnValueConfigurationExtensions.cs" />
+    <Compile Include="../FakeItEasy/ValueTaskReturnValueConfigurationExtensions.StronglyTyped.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\FakeItEasy\FakeItEasy.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'netstandard2.1'">
+    <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.3" />
+  </ItemGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'netstandard1.6'">
+    <DefineConstants>$(DefineConstants);FEATURE_NETCORE_REFLECTION</DefineConstants>
+  </PropertyGroup>
+
+  <!-- Include PDB in package -->
+  <ItemGroup>
+    <None Include="$(OutputPath)/net45/$(AssemblyName).pdb" Pack="true" PackagePath="lib/net45" Visible="false" />
+    <None Include="$(OutputPath)/netstandard1.6/$(AssemblyName).pdb" Pack="true" PackagePath="lib/netstandard1.6" Visible="false" />
+    <None Include="$(OutputPath)/netstandard2.0/$(AssemblyName).pdb" Pack="true" PackagePath="lib/netstandard2.0" Visible="false" />
+  </ItemGroup>
+
+  <!-- Exclude files from netstandard2.1 folder in package -->
+  <Target Name="RemoveNetStd21Files" AfterTargets="_WalkEachTargetPerFramework">
+    <ItemGroup>
+      <_BuildOutputInPackage Remove="$(MSBuildThisFileDirectory)$(IntermediateOutputPath)netstandard2.1/$(AssemblyName).*" Pack="false" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Include a placeholder file in netstandard2.1 folder in package -->
+
+  <Target Name="AddNetStd21Placeholder" BeforeTargets="_WalkEachTargetPerFramework">
+    <WriteLinesToFile File="$(IntermediateOutputPath)_._" Overwrite="true" Lines="" />
+    <ItemGroup>
+      <None Include="$(IntermediateOutputPath)_._" Pack="true" PackagePath="lib/netstandard2.1" Visible="false" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Warning to users targeting netstandard2.1 that this package is no longer required -->
+  <ItemGroup>
+    <None Include="build/FakeItEasy.Extensions.ValueTask.targets" Pack="true" PackagePath="build/netstandard2.1" />
+  </ItemGroup>

```

Comment:
Same approach we used in FakeItEasy.Analyzer. It's intended for users who upgrade their project to netcoreapp3.0, to warn them that they no longer need the package.

I'm not sure that's actually necessary. Without this, they will just keep the package, which will have no effect.

Reply:
Actually... I did a quick check, and it doesn't work on a project that targets multiple TFMs including netstandard2.1. The warning appears because of netstandard2.1, even though the package is necessary for the other TFMs.

I can add a condition on the target that emits the warning, like this:
```
Condition="'$(TargetFrameworks)' == '' or '$(TargetFrameworks)' == '$(TargetFramework)'"
```
It seems to work on a basic scenario, but it the project has multiple TFMs supporting netstandard2.1 (currently only netcoreapp3.0 supports it).

So, either I make that fix, which isn't foolproof, or I just remove the warning completely. What do you prefer?

Reply:
I don't entirely understand. Partly I think due to some minor typos in your comment, and mostly because I would've been barely on the edge of understanding to begin with. I'm getting the gist that sometimes emitting a warning will not work. I say if we're at all unsure about what circumstances will properly trigger the warning, we're better off without, and we can just provide a good description/docs of the package.

Reply:
Sorry, there was a typo and an incomplete sentence, I fixed it (I think).

Maybe an example will help. Suppose I have a test project that targets net472 and netcoreapp3.0. If I reference FakeItEasy.Extensions.ValueTask in that project, I will see the warning, because netcoreapp3.0 doesn't need the package. But net472 _does_ need it, so we shouldn't emit the warning.

Now, with the fix I suggest:
- it will work as expected (i.e. no warning) if my test project still targets net472 and netcoreapp3.0
- BUT if I change my test project to target netcoreapp3.0 and netcoreapp3.1, it won't show the warning, even though it should (because neither netcoreapp3.0 nor netcoreapp3.1 need the package)

So, as I said, the fix isn't foolproof.

> I say if we're at all unsure about what circumstances will properly trigger the warning, we're better off without, and we can just provide a good description/docs of the package.

Sounds good to me.